### PR TITLE
[SHELL32] Watch for common desktop and SHCNE_CREATE for IShellLink::Save

### PR DESCRIPTION
**Pull request by katahiromz**
## Purpose

JIRA issue: [CORE-10391](https://jira.reactos.org/browse/CORE-10391)

On desktop view, we have to watch both the common desktop and the private desktop.
In Windows, `IShellLink::Save` (shortcut creation) sends `SHCNE_CREATE` or `SHCNE_UPDATEITEM` notification.
Simplify `CChangeNotify::ShouldNotify`.

AFTER:
after

### CODE REVIEW

File: dll/win32/shell32/CDefView.cpp [hunk 1]
```diff
@@ -1157,11 +1157,36 @@ LRESULT CDefView::OnCreate(UINT uMsg, WPARAM wParam, LPARAM lParam, BOOL &bHandl
...
+    INT nRegCount;
+    SHChangeNotifyEntry ntreg[3];
+    PIDLIST_ABSOLUTE pidls[3];
+    if ((m_FolderSettings.fFlags & FWF_DESKTOP) || _ILIsDesktop(m_pidlParent))
+    {
+        nRegCount = 3;
+        SHGetSpecialFolderLocation(m_hWnd, CSIDL_DESKTOPDIRECTORY, &pidls[0]);
+        SHGetSpecialFolderLocation(m_hWnd, CSIDL_COMMON_DESKTOPDIRECTORY, &pidls[1]);
+        SHGetSpecialFolderLocation(m_hWnd, CSIDL_BITBUCKET, &pidls[2]);
+        ntreg[0].fRecursive = TRUE;
```
Comment (learn-more) [inline dll/win32/shell32/CDefView.cpp:1169]: Do we need recursive watch on the desktop?
(I don't know the answer btw, just wondering)

File: dll/win32/shell32/CShellLink.cpp [hunk 2]
```diff
@@ -357,6 +362,12 @@ HRESULT STDMETHODCALLTYPE CShellLink::Save(LPCOLESTR pszFileName, BOOL fRemember
 
         if (SUCCEEDED(hr))
         {
+            if (!bAlreadyExists)
+            {
+                GetFullPathNameW(pszFileName, _countof(szFullPath), szFullPath, NULL);
+                SHChangeNotify(SHCNE_CREATE, SHCNF_PATHW, szFullPath, NULL);
+            }
```
Comment (learn-more) [inline dll/win32/shell32/CShellLink.cpp:369]: Is it needed to send an `SHCNE_UPDATEITEM` in the else case?

Comment (katahiromz) [reply dll/win32/shell32/CShellLink.cpp]: yeah
Yeah. OK.

File: dll/win32/shell32/CDefView.cpp (hunk 1 above)
Comment (katahiromz) [reply dll/win32/shell32/CDefView.cpp]: Non-recursive is better. OK.

Comment (katahiromz) [reply dll/win32/shell32/CDefView.cpp]: no-error
Still OK.

**Comment by katahiromz** [comment]
Still OK.
ok1
OK.
ok2
Ready to go.

**Pull request merged**